phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -1562,7 +1678,7 @@ private MetaDataMutationResult doDropTable(byte[] key, byte[] tenantId, byte[] s
             }
 
             if (tableType != PTableType.VIEW) { // Add to list of HTables to delete, unless it's a view
-                tableNamesToDelete.add(table.getName().getBytes());
+                tableNamesToDelete.add(table.getPhysicalName().getBytes());
```

Comment:
Yes!! it was not correct earlier but it didn't matter because logical and physical representation was same. but now this change is required to get correct physical name so that we can delete descriptors on the client.